src/Microsoft.AspNet.Mvc.Core/MvcOptions.cs

```diff
@@ -22,16 +22,16 @@ public class MvcOptions
         public MvcOptions()
         {
             Conventions = new List<IApplicationModelConvention>();
-            ModelBinders = new List<ModelBinderDescriptor>();
+            ModelBinders = new List<IModelBinder>();
             ViewEngines = new List<ViewEngineDescriptor>();
```

Comment:
A bunch of stuff it uses is currently scoped. We should simplify the rest of these and then find the right solution for the view engine by looking at all the components and what needs to be stateful and why (not as part of this PR).